### PR TITLE
Some fixes and improvements for Loom Debugger

### PR DESCRIPTION
**Pull request by themuflon**

### CODE REVIEW

File: loom/script/native/core/system/lmDebug.cpp [hunk 1]
```diff
@@ -858,42 +861,104 @@ class Debug {
 
     // add's a breakpoint at the given source and line, checks for duplicates
     // and avoids them
-    static void addBreakpoint(const char *source, int line)
+    //static const char* addBreakpoint(const char *source, int line)
```
Comment (SmilyOrg) [inline loom/script/native/core/system/lmDebug.cpp]: Dead code?

File: loom/script/native/core/system/lmDebug.cpp [hunk 2]
```diff
@@ -858,42 +861,104 @@ class Debug {
...
+                return 1;
+            }
+        }
+        lua_pushstring(L, NULL);
+        return 1;
     }
 
     // removes a breakpoint at the given source and line
-    static void removeBreakpoint(const char *source, int line)
+    //static const char* removeBreakpoint(const char *source, int line)
```
Comment (SmilyOrg) [inline loom/script/native/core/system/lmDebug.cpp]: Dead code?

File: sdk/src/ldb/LDB.ls [hunk 3]
```diff
@@ -53,47 +53,47 @@ package ldb {
...
-        
+
         public function doSomething() {
             trace("hi");
         }
-        
+
     }
-    
-
```
Comment (SmilyOrg) [inline sdk/src/ldb/LDB.ls]: What's with all the whitespace changes here? @bengarney what's our whitespace-on-empty-line policy?

File: sdk/src/system/Debug.ls [hunk 4]
```diff
@@ -93,55 +93,55 @@ package system {
...
          * @see CallStackInfo
          */
         public static native function getCallStack():Vector.<CallStackInfo>;
-        
+
         /*
          * Adds a breakpoint at the given source and line number.
          */
-        public static native function addBreakpoint(source:String, line:Number);
+        public static native function addBreakpoint(source:String, line:Number):String;
```
Comment (SmilyOrg) [inline sdk/src/system/Debug.ls]: What does it return now? Can you document it for this one and the ones below?

File: sdk/src/system/Debugger/Client.ls [hunk 5]
```diff
@@ -110,12 +110,12 @@ package system.debugger
          *  at an initialized state in the application.  Debugging commands such as
          *  stepping and object inspecting do work before reloading is enabled.
          */
-        static function cmdRELOAD() { 
+        static function cmdRELOAD():Boolean {
```
Comment (SmilyOrg) [inline sdk/src/system/Debugger/Client.ls]: Can you document what this function returns? Same for the ones changed below.

File: sdk/src/system/Debugger/Client.ls [hunk 6]
```diff
@@ -130,210 +130,226 @@ package system.debugger
...
-            Debug.addBreakpoint(commandv[1], commandv[2].toNumber());
+            if (checkAssertionState())
+                return false;
 
-            sendToServer("Setting breakpoint: " + commandv[1] + " : " + commandv[2].toNumber() + "\n");                
+            var commandv:Vector.<String> = line.find("^([A-Z]+)%s+(.-)%s+(%d+)%s*$");
 
-        
+            var result:String;
+            if((result = Debug.addBreakpoint(commandv[1], commandv[2].toNumber())) != null)
```
Comment (SmilyOrg) [inline sdk/src/system/Debugger/Client.ls]: Can this line crash with out-of-bounds because of the regex above failing to match all the groups?

File: sdk/src/system/Debugger/Client.ls [hunk 7]
```diff
@@ -130,210 +130,226 @@ package system.debugger
...
-        static function cmdCLEAR(line:String) {
-        
+        static function cmdCLEAR(line:String):Boolean {
+
             if (checkAssertionState())
-                return;
-        
+                return false;
+
             var commandv:Vector.<String> = line.find("^([A-Z]+)%s+(.-)%s+(%d+)%s*$");
```
Comment (SmilyOrg) [inline sdk/src/system/Debugger/Client.ls]: If this regex is the same as the one above, maybe we could have it up top as a constant string matcher?

File: sdk/src/system/Debugger/Server.ls [hunk 8]
```diff
@@ -14,245 +14,264 @@ Unless required by applicable law or agreed to in writing, software
...
                 s = client.receive();
-            } 
-            
+            }
+
             var rbuffer = lines.join("\n");
             if (!rbuffer.length)
-                rbuffer = "OK";
-            
+                rbuffer = null;
```
Comment (SmilyOrg) [inline sdk/src/system/Debugger/Server.ls]: Are all the functions that call this one null-aware now?

**Comment by SmilyOrg** [comment]
A few stylistic and doc nits mostly. Would be nicer if all the whitespace changes were in a separate PR.

File: sdk/src/system/Debugger/Server.ls (hunk 8 above)
Comment (themuflon) [inline sdk/src/system/Debugger/Server.ls]: They are.

**Comment by themuflon** [comment]
Nitted

File: sdk/src/ldb/LDB.ls (hunk 3 above)
Comment (bengarney) [inline sdk/src/ldb/LDB.ls]: A couple whitespace changes are OK, but we shouldn't make more than say 3-4 without a good reason. We want to match the standards of the file we are in.

Comment (SmilyOrg) [inline sdk/src/ldb/LDB.ls]: @themuflon can you revert the whitespace changes and put up a separate PR if the changes would make it more consistent with other files?

**Comment by SmilyOrg** [comment]
Github's diff looks broken, but I think this is fine now, landing!
